### ISSUE TITLE
Code with syntax errors is removed on first save

### ISSUE THREAD

**Issue by lukaszcz** [anoma/vscode-juvix#34]
If you type in
```agda
module Bug;
end;
```
save, then change it to
```agda
module Bug;

f : Nat -> Nat;
f x := x;

end;
```
and save, then the module body disappears with no error message and you get
```agda
module Bug;
end;
```
back again.

**Comment by lukaszcz** [comment]
Well, it actually happens even with well-typed code. Adding `open import Stdlib.Prelude` doesn't help. You need to manually click on the "typecheck" button first.

**Comment by lukaszcz** [comment]
And then your code gets automatically reformatted! This alone should be considered a bug - not everyone wants their code reformatted on every save.

**Comment by lukaszcz** [comment]
The VSCode extension is practically unusable. Now it sometimes randomly replaces the newly written code with the old version!

**Comment by jonaprieto** [comment]
Do you have the option "Format on save?"

**Comment by jonaprieto** [comment]
I suspect this issue has to be with the Juvix formatting. Did you experience this issue before 0.1.22?

**Comment by lukaszcz** [comment]
No. I use default options, didn't change anything, only updated the
extension.

**Comment by lukaszcz** [comment]
Maybe I have global "format on save". I'll check that.

**Comment by jonaprieto** [comment]
Let me see if I can experience the same, because I can't locally.

**Comment by jonaprieto** [comment]
I couldn't reproduce the behavior you described. :(

**Comment by lukaszcz** [comment]
I have "format on save" turned on. You can't change it locally, it seems, and I don't want to turn it off globally. The Juvix formatter doesn't work the way it should, removing text it cannot parse. The easiest fix is to just print out the input if there is a parse error.

**Comment by lukaszcz** [comment]
And it's not only incorrect code. Sometimes perfectly valid code gets removed by the formatter. See my second comment above.

**Comment by jonaprieto** [comment]
It was fixed in v0.1.24. Reopen if you experience the same behaviour.
